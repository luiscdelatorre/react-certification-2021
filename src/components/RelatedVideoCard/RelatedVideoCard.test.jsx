import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import RecommendedCard from './RelatedVideoCard.component';
import relatedVideos from '../../mock/youtube-related-videos-mock.json';

describe('Related Video Card component', () => {
  it('Should display mock data in component', () => {
    const video = relatedVideos.items[0];
    const { title, channelTitle, thumbnail, liveBroadcastContent } = video.snippet;
    const { id } = video;
    render(
      <BrowserRouter>
        <RecommendedCard
          id={id.videoId}
          title={title}
          channel={channelTitle}
          thumbnail={thumbnail}
          liveBroadcastContent={liveBroadcastContent}
        />
      </BrowserRouter>
    );

    expect(screen.getByText(video.snippet.title).tagName).toBe('H5');
    expect(screen.getByText(video.snippet.channelTitle).tagName).toBe('H6');
  });
  it('Should render conditional item for liveBroadcastContent', () => {
    const video = relatedVideos.items[0];
    const { title, channelTitle, thumbnail, liveBroadcastContent } = video.snippet;
    const { id } = video;
    render(
      <BrowserRouter>
        <RecommendedCard
          id={id.videoId}
          title={title}
          channel={channelTitle}
          thumbnail={thumbnail}
          liveBroadcastContent={liveBroadcastContent}
        />
      </BrowserRouter>
    );

    expect(screen.getByText(video.snippet.title).tagName).toBe('H5');
    expect(screen.getByText(video.snippet.channelTitle).tagName).toBe('H6');
    expect(screen.getByText('live')).toBeInTheDocument();
  });

  it('Validates VideoCard snapshot', () => {
    const video = relatedVideos.items[0];
    const { title, channelTitle, thumbnail, liveBroadcastContent } = video.snippet;
    const { id } = video;
    const component = renderer.create(
      <BrowserRouter>
        <RecommendedCard
          id={id.videoId}
          title={title}
          channel={channelTitle}
          thumbnail={thumbnail}
          liveBroadcastContent={liveBroadcastContent}
        />
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
