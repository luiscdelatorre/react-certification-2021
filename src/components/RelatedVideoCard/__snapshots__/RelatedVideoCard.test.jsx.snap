// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Related Video Card component Validates VideoCard snapshot 1`] = `
<a
  className="sc-bdfBwQ iEkQBl"
  href="/video/_0fktNnvy4M"
  onClick={[Function]}
>
  <header
    className="sc-gsTCUz gbEMKB"
  >
    <img
      alt="Ash & Pikachu's Epic Battle Moments | Pokémon Journeys | Netflix Futures"
    />
    <h6
      className="sc-hKgILt cThFfx"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M346.65 304.3a136 136 0 00-180.71 0 21 21 0 1027.91 31.38 94 94 0 01124.89 0 21 21 0 0027.91-31.4z"
        />
        <path
          d="M256.28 183.7a221.47 221.47 0 00-151.8 59.92 21 21 0 1028.68 30.67 180.28 180.28 0 01246.24 0 21 21 0 1028.68-30.67 221.47 221.47 0 00-151.8-59.92z"
        />
        <path
          d="M462 175.86a309 309 0 00-411.44 0 21 21 0 1028 31.29 267 267 0 01355.43 0 21 21 0 0028-31.31z"
        />
        <circle
          cx="256.28"
          cy="393.41"
          r="32"
        />
      </svg>
      live
    </h6>
  </header>
  <div
    className="sc-dlfnbm fOolBg"
  >
    <h5
      className="sc-eCssSg eMkQAo"
      title="Ash & Pikachu's Epic Battle Moments | Pokémon Journeys | Netflix Futures"
    >
      Ash & Pikachu's Epic Battle Moments | Pokémon Journeys | Netflix Futures
    </h5>
    <h6
      className="sc-jSgupP hHNFVW"
    >
      Netflix Futures
    </h6>
  </div>
</a>
`;
